### PR TITLE
[do not merge] Edits procedures in builds file

### PR DESCRIPTION
**Pull request by bfallonf**
@soltysh This PR is in response to your PR: https://github.com/openshift/openshift-docs/pull/339

I have some questions, though it might be easier to ask them in the file as they're about specific parts that would be hard to reference here...

Thanks!

### CODE REVIEW

File: architecture/core_objects/builds.adoc [hunk 1]
```diff
@@ -169,24 +170,22 @@ following format:
...
-command. The file will be created if it does not exist.
-
-Kubernetes provides the
-https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/secrets.md[Secret]
-resource, which is used to store your configuration and passwords. You must
-first create the `*Secret*` before builds can use your *_.dockercfg_* file for
-pushing the output image:
+command. The file will be created if it does not exist. Kubernetes provides
+https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/secrets.md[Secrets],
+which are used to store your configuration and passwords.
```
Comment (bfallonf) [inline architecture/core_objects/builds.adoc]: @soltysh If Kubernetes provides the secret, why are we creating one in the next step? Would it more accurate to say something like "Kubernetes _has the ability to utilize_ secret..." then we go on to create the secret for Kubernetes to utilize?

File: architecture/core_objects/builds.adoc [hunk 2]
```diff
@@ -169,24 +170,22 @@ following format:
...
 ----
 ====
 
-The `*data*` field for the `*Secret*` object must contain the `*dockercfg*` key
-with the value set to the base64-encoded content of the *_.dockercfg_* file. For
-example:
-
+. The `*data*` field in the `*Secret*` must contain the `*dockercfg*` value set
+to the base64-encoded content of the *_.dockercfg_* file. For example:
++
```
Comment (bfallonf) [inline architecture/core_objects/builds.adoc]: This question might need to be reference in the SSH procedure as well, but is this missing a step? In the SSH procedure we use the `base64` command to find a value to put in the config file. Should a similar step be here?

File: architecture/core_objects/builds.adoc [hunk 3]
```diff
@@ -204,10 +203,10 @@ example:
...
 ====
 
-Once you have the `*Secret*` created, you can add a `PushSecret` field into the
-`Output` section of the `BuildConfig` and set it to the name of the `*Secret*`
-that you created, which in the above example is `*dockerhub*`:
-
+. Add a `PushSecret` field to the `output` section of the `BuildConfig` and
+set it to the name of the `*Secret*` created in step one. As in the above
+example, the following uses `*dockerhub*`:
++
```
Comment (bfallonf) [inline architecture/core_objects/builds.adoc]: This is straight-forward, and I can see the name in the above step too, but when is the name established? In the above step too?

File: architecture/core_objects/builds.adoc [hunk 4]
```diff
@@ -223,9 +222,9 @@ that you created, which in the above example is `*dockerhub*`:
 ----
 ====
 
-Pull the builder Docker image from a private Docker registry by specifying the
+. Pull the Docker build image from a private Docker registry by specifying the
 `PullSecret` field, which is part of the build strategy definition:
-
++
```
Comment (bfallonf) [inline architecture/core_objects/builds.adoc]: Which file is this in? The buildconfig?

File: architecture/core_objects/builds.adoc [hunk 5]
```diff
@@ -302,19 +310,19 @@ Copy the value returned from the above command and place it into the
 ----
 ====
```
Comment (bfallonf) [inline architecture/core_objects/builds.adoc]: Similar to the question in the Docker procedure above, I just want to clarify this is the correct procedure. We find the value using the `base64` command, then enter it into the secret.json file. Correct?

File: architecture/core_objects/builds.adoc [hunk 6]
```diff
@@ -302,19 +310,19 @@ Copy the value returned from the above command and place it into the
...
 
-Then, create the `*Secret*` from the *_secret.json_* file using the following
+. Create the `*Secret*` from the *_secret.json_* file using the following
```
Comment (bfallonf) [inline architecture/core_objects/builds.adoc]: Why is another secret being created here?

File: architecture/core_objects/builds.adoc [hunk 7]
```diff
@@ -302,19 +310,19 @@ Copy the value returned from the above command and place it into the
...
 ====
 ----
 $ osc create -f secret.json
 ----
 ====
 
-Add a `SourceSecret` field into the `Source` section inside the `BuildConfig`
+. Add a `SourceSecret` field into the `Source` section inside the `BuildConfig`
 and set it to the name of the `*Secret*` that you created, in this case
 `*scmsecret*`:
```
Comment (bfallonf) [inline architecture/core_objects/builds.adoc]: Similar to the Docker procedure above, when and how was the name of the secret established?

**Comment by adellape** [comment]
@bfallonf A lot of similar changes going on in https://github.com/openshift/openshift-docs/pull/431

**Comment by bfallonf** [comment]
@adellape True. Similar, but not exact. Once the other merges I'll get that info in here and go through this one again. I think most (all?) of my questions above still stand.

**Comment by adellape** [comment]
@bfallonf Yea, my comment here was just an FYI that this would probably conflict. Merged 431, so this needs a rebase.

**Comment by soltysh** [comment]
@bfallonf sorry, I was out yesterday, will review this today.

File: architecture/core_objects/builds.adoc (hunk 1 above)
Comment (soltysh) [inline architecture/core_objects/builds.adoc]: Yes, by provides I've meant about the ability to create ones.

File: architecture/core_objects/builds.adoc (hunk 2 above)
Comment (soltysh) [inline architecture/core_objects/builds.adoc]: Yeah, it should simplify user experience.

File: architecture/core_objects/builds.adoc (hunk 3 above)
Comment (soltysh) [inline architecture/core_objects/builds.adoc]: This is this part of the secret.json file:
```
"metadata": {
    "name": "dockerhub"
  },
```

File: architecture/core_objects/builds.adoc (hunk 4 above)
Comment (soltysh) [inline architecture/core_objects/builds.adoc]: Yes, this is part of the BuildConfig specification, this part defines the strategy used during build.

File: architecture/core_objects/builds.adoc (hunk 5 above)
Comment (soltysh) [inline architecture/core_objects/builds.adoc]: Yup.

File: architecture/core_objects/builds.adoc (hunk 6 above)
Comment (soltysh) [inline architecture/core_objects/builds.adoc]: SourceSecrets are separate topic from DockerSecrets, thus we create another secret. You can use both, but we've decided to have each topic covered full path.

File: architecture/core_objects/builds.adoc (hunk 7 above)
Comment (soltysh) [inline architecture/core_objects/builds.adoc]: See above.

**Comment by soltysh** [comment]
@bfallonf you should be ok, be aware though we'll be moving this file contents into dev_guide/builds.adoc, heads up for #444.

**Comment by bfallonf** [comment]
Seems #444 has created many more conflicts. I'll get to this once that's ironed out.

**Comment by adellape** [comment]
@bfallonf Should be safe to rebase now.

**Comment by bfallonf** [comment]
Might have to close this one. The builds file has been through too many changes to make this workable. The work from this PR is now in  #496

**Pull request closed without merging**